ansible/roles/piwik/templates/nginx-site-piwik.j2

```diff
@@ -0,0 +1,67 @@
+server {
+    server_name {{ server_name}};
+
+    listen 443 ssl default_server;
+    listen [::]:443 ssl default_server;
+
+    ssl_certificate /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/privkey.pem;
+    ssl_trusted_certificate /etc/letsencrypt/live/{{ letsencrypt_domains.split(',')[0] }}/fullchain.pem;
+
+  if ($request_method !~ ^(GET|HEAD|POST)$ ) {
+    return 444;
+  }
+
+  root  {{ app_path }}/current/piwik;
+  index  index.php index.html;
+  # Disallow any usage of piwik assets if referer is non valid.
+  location ~* ^.+\.(?:jpg|png|css|gif|jpeg|js|swf)$ {
+    # Defining the valid referers.
+    #valid_referers none blocked *.mysite.com othersite.com;
+
+    expires max;
```

Comment:
Maybe E-Tag may actually not be enough. However I don't see this as being fatal, it just means that if we make an update to piwik the client side code will not be updated in browser that have piwik.js already cached. I don't think that is the end of the world to be honest as I would expect the client code to be backward compatible. In any case we can deploy cache breaking solution (as outlined above) without having to redeploy on another domain if we care about this.